nltk/parse/dependencygraph.py

```diff
@@ -244,48 +296,75 @@ def triples(self, node=None):
             for triple in self.triples(node=dep):
                 yield triple
 
-
     def _hd(self, i):
         try:
-            return self.nodelist[i]['head']
+            return self.nodes[i]['head']
         except IndexError:
             return None
 
     def _rel(self, i):
         try:
-            return self.nodelist[i]['rel']
+            return self.nodes[i]['rel']
         except IndexError:
             return None
 
     # what's the return type?  Boolean or list?
     def contains_cycle(self):
+        """Check whether there are cycles.
+
+        >>> dg = DependencyGraph(treebank_data)
+        >>> dg.contains_cycle()
+        False
+
+        >>> cyclic_dg = DependencyGraph()
+        >>> top = {'word': None, 'deps': [1], 'rel': 'TOP', 'address': 0}
+        >>> child1 = {'word': None, 'deps': [2], 'rel': 'NTOP', 'address': 1}
+        >>> child2 = {'word': None, 'deps': [4], 'rel': 'NTOP', 'address': 2}
+        >>> child3 = {'word': None, 'deps': [1], 'rel': 'NTOP', 'address': 3}
+        >>> child4 = {'word': None, 'deps': [3], 'rel': 'NTOP', 'address': 4}
+        >>> cyclic_dg.nodes = {
+        ...     0: top,
+        ...     1: child1,
+        ...     2: child2,
+        ...     3: child3,
+        ...     4: child4,
+        ... }
+        >>> cyclic_dg.root = top
+
+        >>> cyclic_dg.contains_cycle()
+        [3, 1, 2, 4]
+
+        """
         distances = {}
-        for node in self.nodelist:
+
+        for node in self.nodes.values():
             for dep in node['deps']:
-                key = tuple([node['address'], dep]) #'%d -> %d' % (node['address'], dep)
+                key = tuple([node['address'], dep])
                 distances[key] = 1
-        for n in range(len(self.nodelist)):
+
+        for _ in self.nodes:
             new_entries = {}
+
             for pair1 in distances:
                 for pair2 in distances:
                     if pair1[1] == pair2[0]:
                         key = tuple([pair1[0], pair2[1]])
                         new_entries[key] = distances[pair1] + distances[pair2]
+
             for pair in new_entries:
                 distances[pair] = new_entries[pair]
                 if pair[0] == pair[1]:
-                    print(pair[0])
-                    path = self.get_cycle_path(self.get_by_address(pair[0]), pair[0]) #self.nodelist[pair[0]], pair[0])
+                    path = self.get_cycle_path(self.get_by_address(pair[0]), pair[0])
                     return path
-        return False  # return []?
 
+        return False  # return []?
```

Comment:
Why not return [] indeed? The method can be renamed e.g. to get_cycles then. For backwards compatibility (if it is desired) we can add something like

```
def contains_cycle(self):
    return self.get_cycles() or False
```